quipucords/scanner/input_log.py

```diff
@@ -0,0 +1,301 @@
+# Copyright (c) 2018 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+
+"""Log all input to the filesystem for later debugging."""
+
+import json
+import logging
+import multiprocessing as mp
+import os
+import time
+import uuid
+from django.db import transaction
+from api import log_state
+from quipucords import settings
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+UUID_CACHE = None
+
+DEFAULT_MAX_SIZE = 1 << 30  # 1 GiB
+DEFAULT_MAX_AGE = 365 * 24 * 60 * 60  # 1 year, except on leap years
+
+
+@transaction.atomic()
+def get_sonar_uuid():
+    """Get the UUID of this Sonar installation."""
+    global UUID_CACHE  # pylint: disable=global-statement
+    if UUID_CACHE is not None:
+        return UUID_CACHE
+
+    uuid_obj = log_state.DatabaseUUID.objects.all().first()
+    if uuid_obj is None:
+        uuid_obj = log_state.DatabaseUUID(uuid=uuid.uuid4())
+        uuid_obj.save()
+    UUID_CACHE = uuid_obj.uuid
+
+    return uuid_obj.uuid
+
+
+# This is a potential performance bottleneck since we are touching the
+# database every time we write a record. This isn't necessary - we
+# could get the state of the counter by reading the last few entries
+# of the existing log files on startup. I don't plan to implement that
+# unless we find that this is actually a problem.
+@transaction.atomic()
```

Comment:
What is all the UUID stuff providing us at this point? I can only see it making it harder to compress the file since it will be filled with UUIDs. I know down the line you want it for replaying, but I think things will likely change to much between now and then and even this will not be backwards compatible with where we end up.

Reply:
Fair point. The UUIDs in the log file will all be the same, so they should compress pretty well. We don't need the UUIDs for replaying; they're there to make sure that we don't accidentally mix records from two different installations on our end. As long as they have different UUIDs, we can at least notice that the UUIDs don't match.